subsys/zigbee/cli/zigbee_cli_cmd_nbr.c

```diff
@@ -0,0 +1,262 @@
+{
+	zb_uint32_t i;
+	char ieee_addr_buf[IEEE_ADDR_BUF_SIZE] = { 0 };
+	int addr_len;
+
+	LOG_INF("Active neighbor table (%d entries):", nbr_cache.n_entries);
+	LOG_INF("------------------------------");
+
+	for (i = 0; i < nbr_cache.n_entries; i++) {
+		addr_len =
+			ieee_addr_to_str(ieee_addr_buf,
+					 sizeof(ieee_addr_buf),
+					 nbr_cache.entries[i].ieee_addr);
+		if (addr_len < 0) {
+			strcpy(ieee_addr_buf, "unknown");
+		}
+
+		ZB_ASSERT(nbr_cache.entries[i].device_type < 4);
+		ZB_ASSERT(nbr_cache.entries[i].relationship < 6);
+
+		LOG_INF("ieee_addr: %s", log_strdup(ieee_addr_buf));
+		LOG_INF("  device_type: %s", device_type_name[nbr_cache.entries[i].device_type]);
+		LOG_INF("  relationship: %s", relationship_name[nbr_cache.entries[i].relationship]);
+		LOG_INF("  outgoing_cost: %hd", nbr_cache.entries[i].outgoing_cost);
+		if (nbr_cache.entries[i].device_type == 2) {
+			LOG_INF("  timeout_counter: %d", nbr_cache.entries[i].timeout_counter);
+		} else {
+			LOG_INF("  age: %hd", nbr_cache.entries[i].age);
+		}
+	}
+
+	LOG_INF("------------------------------");
+}
+
+static void refresh_active_nbt_table_delayed(zb_bufid_t bufid)
+{
+	zb_ret_t error_code;
+
+	error_code = zb_nwk_nbr_iterator_next(bufid, refresh_active_nbt_table);
+	ZB_ASSERT(error_code == RET_OK);
+}
+
+static void refresh_active_nbt_table(zb_bufid_t bufid)
+{
+	static zb_bool_t cache_updated = ZB_FALSE;
+	zb_ret_t error_code;
+	zb_nwk_nbr_iterator_params_t *args = ZB_BUF_GET_PARAM(bufid, zb_nwk_nbr_iterator_params_t);
+	zb_nwk_nbr_iterator_entry_t *entry = (zb_nwk_nbr_iterator_entry_t *)zb_buf_begin(bufid);
+
+	/* Initialize the params structure if this is the first API call. */
+	if (last_update_count == UPDATE_COUNT_RELOAD) {
+		args->update_count = 0;
+		args->index = ZB_NWK_NBR_ITERATOR_INDEX_EOT;
+		zb_buf_set_status(bufid, RET_OK);
+	}
+
+	error_code = zb_buf_get_status(bufid);
+	ZB_ASSERT(error_code == RET_OK);
+
+	if (args->update_count != last_update_count) {
+		/* Counter skew - restart the read process. */
+		active_nbr_clear();
+		args->index = 0;
+		last_update_count = args->update_count;
+		cache_updated = ZB_TRUE;
+	} else if (args->index != ZB_NWK_NBR_ITERATOR_INDEX_EOT) {
+		/* Counter correct, regular entry - add it to the cache. */
+		active_nbr_store(entry);
+		args->index++;
+		cache_updated = ZB_TRUE;
+	} else if (cache_updated) {
+		/* Counter correct, last entry received.
+		 * Previous call contained a regular entry - log the table.
+		 */
+		active_nbr_log();
+		cache_updated = ZB_FALSE;
+	}
+
+	if (args->index != ZB_NWK_NBR_ITERATOR_INDEX_EOT) {
+		/* Regular entry received - call the API immediately. */
+		error_code = zb_nwk_nbr_iterator_next(bufid, refresh_active_nbt_table);
+		ZB_ASSERT(error_code == RET_OK);
+	} else if (monitor_active == ZB_TRUE) {
+		/* Regular entry not present - call the API after 1 second. */
+		ZB_SCHEDULE_APP_ALARM(
+			refresh_active_nbt_table_delayed,
+			bufid,
+			ZB_TIME_ONE_SECOND);
```

Comment:
What do you think about making this refresh period configurable? Would it be worth it?